pkg/snapstore/s3_snapstore.go

```diff
@@ -495,30 +495,61 @@ func (s *S3SnapStore) Delete(snap brtypes.Snapshot) error {
 	return err
 }
 
-// S3SnapStoreHash calculates and returns the hash of aws S3 snapstore secret.
-func S3SnapStoreHash(config *brtypes.SnapstoreConfig) (string, error) {
+// S3SnapStoreModTime returns the last modification time of the credential files
+func S3SnapStoreModTime(config *brtypes.SnapstoreConfig) (time.Time, error) {
 	if dir, isSet := os.LookupEnv(awsCredentialFile); isSet {
-		awsConfig, err := readAWSCredentialFromDir(dir)
+		awsTimeStamp, err := readAWSCredentialModTime(dir)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials from %v directory", dir)
+			return time.Time{}, fmt.Errorf("error getting credentials using %v directory with error: %v", dir, err.Error())
 		}
-		return getS3Hash(awsConfig), nil
+		return awsTimeStamp, nil
 	}
 
 	if filename, isSet := os.LookupEnv(awsCredentialJSONFile); isSet {
-		awsConfig, err := credentialsFromJSON(filename)
+		fileInfo, err := os.Stat(filename)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials using %v file", filename)
+			return time.Time{}, fmt.Errorf("error getting json credentials %v with error %v", filename, err.Error())
```

Comment:
```suggestion
			return nil, fmt.Errorf("failed to get AWS credentials from JSON file '%v': %v", filename, err)
```

Reply:
Applied error formatting suggestion to all occurrences.